jsx-factory.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import('./lib/core.js').Element} Element
+ * @typedef {import('./lib/core.js').Root} Root
+ * @typedef {import('./lib/core.js').HResult} HResult
+ * @typedef {import('./lib/core.js').HArrayChild} HArrayChild
+ * @typedef {import('./lib/core.js').HProperties} HProperties
+ * @typedef {import('./lib/core.js').HPropertyValue} HPropertyValue
+ * @typedef {import('./lib/core.js').HStyle} HStyle
+ * @typedef {import('./lib/core.js').core} Core
+ *
+ * @typedef {Record<string, unknown>} JSXProps
+ * @typedef {(props: JSXProps) => HResult } HFn
+ */
+
+import {h as hast} from './index.js'
+
+/**
+ * @param {string | null | HFn} typeOrFn
+ * @param { JSXProps | undefined } props
+ * @param { HArrayChild } children
+ * @returns HResult
```

Comment:
```suggestion
 * @param {HFn | string | null} typeOrFn
 *   Component, tag name, or `null` to generate a `Root`.
 * @param {JSXProps | null | undefined} props
 *   Properties.
 * @param {HArrayChild} children
 *   Children.
 * @returns HResult
 *   Result.
```

Sorting, docs, no superfluous whitespace

Reply:
maybe just call it `component` instead of `typeOrFn`?